### PR TITLE
[Keyboard] Added Compound keyboard support

### PR DESCRIPTION
**Pull request by yumagulovrn**
## Description

Added support for the Compound keyboard, which soon will be available via GB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/compound/compound.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{ \
+    {k00, k01, k02, k03, k04, k05, k06, k07, k08, k09, k0a, k0b, k0c, k0d, k0e}, \
+    {k10, k11, k12, k13, k14, k15, k16, k17, k18, k19, k1a, k1b, k1c, k1d, XXX}, \
+    {k20, k21, k22, k23, k24, k25, k26, k27, k28, k29, k2a, k2b, k2c, XXX, XXX}, \
+    {k30, k31, k32, k33, k34, k35, k36, k37, k38, k39, k3a, k3b, k3c, XXX, XXX}, \
+    {k40, k41, k42, XXX, XXX, XXX, k46, XXX, XXX, XXX, k4a, k4b, k4c, XXX, XXX}  \
+}
+
+void matrix_init_user(void);
+void matrix_scan_user(void);
```
Comment (drashna) [inline keyboards/compound/compound.h:41]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/compound/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+![Compound](https://imgur.com/0DC9F9d.jpg)
+
+A compact 60% Compound keyboard with fixed Tsangan style layout.
+
+* Keyboard Maintainer: [yumagulovrn](https://github.com/yumagulovrn)  
+* Hardware Supported: Compound PCB
+* Hardware Availability: [GeekHack IC](https://geekhack.org/index.php?topic=111187.0)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/compound/readme.md:11]: Could you also add instructions on how to reset the controller into the bootloader mode, so new firmware can be flashed?  

Eg, does it have a physical reset button?  Bootmagic?  Reset keycode?

File: keyboards/compound/compound.h (hunk 1 above)
Comment (yumagulovrn) [reply keyboards/compound/compound.h]: Thank you! Removed unnecessary lines

File: keyboards/compound/readme.md (hunk 2 above)
Comment (yumagulovrn) [reply keyboards/compound/readme.md:26]: Thank you for the comment! Added proper instructions in 4a811a7

**Review by drashna — APPROVED** [review]

File: keyboards/compound/rules.mk [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+MOUSEKEY_ENABLE = no    # Mouse keys
+EXTRAKEY_ENABLE = yes   # Audio control and System control
+CONSOLE_ENABLE = no	    # Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = no        # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = no    # Enable keyboard underlight functionality
+BACKLIGHT_ENABLE = no   # Enable keyboard backlight functionality
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no     # Unicode
+BLUETOOTH_ENABLE = no   # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/compound/rules.mk:18]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/compound/info.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "Compound",
+    "url": "https://geekhack.org/index.php?topic=111187.0",
+    "maintainer": "yumagulovrn",
+    "width": 15,
+    "height": 5,
+    "key_count": 62,
```
Comment (fauxpark) [inline keyboards/compound/info.json:7]:
```suggestion
```

File: keyboards/compound/readme.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Compound
+
+![Compound](https://imgur.com/0DC9F9d.jpg)
```
Comment (fauxpark) [inline keyboards/compound/readme.md:3]:
```suggestion
![Compound](https://imgur.com/0DC9F9dh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**